### PR TITLE
led_strip: support dma feature and various clock source

### PR DESCRIPTION
**Pull request by suda-morris**
This PR enables the user to select more clock sources for the led strip (RMT backend). 

DMA support can also be enabled optionally.

Restructured the component source file and updated the README file.

### CODE REVIEW

**Comment by suda-morris** [comment]
Hi @tore-espressif  would you please help review the PR?

File: led_strip/include/led_strip_rmt.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+/**
+ * @brief LED Strip RMT specific configuration
+ */
+typedef struct {
+    rmt_clock_source_t clk_src; /*!< RMT clock source */
+    uint32_t resolution_hz;     /*!< RMT tick resolution, if set to zero, a default resolution (10MHz) will be applied */
+    struct {
+        uint32_t with_dma: 1;   /*!< Use DMA to transmit data */
```
Comment (tore-espressif) [inline led_strip/include/led_strip_rmt.h:25]: Would it be worth adding more RMT flags? Such as `invert_out` or `io_od_mode`?

(https://github.com/espressif/esp-idf/blob/v5.0/components/driver/include/driver/rmt_tx.h#L44)

**Review by tore-espressif — APPROVED** [review]
@suda-morris Just one question. Otherwise LGTM!

File: led_strip/include/led_strip_rmt.h (hunk 1 above)
Comment (suda-morris) [reply led_strip/include/led_strip_rmt.h:25]: Thanks for the reminder, I added the `nver_out` flag now. But the "io_od_mode" is not necessary for the led_strip application AFAIK.

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**